pysquared/state_of_health.py

```diff
@@ -0,0 +1,100 @@
+from pysquared.config.config import Config
+from pysquared.logger import Logger
+from pysquared.protos.power_monitor import PowerMonitorProto
+
+try:
+    from typing import Callable, List
+
+except Exception:
+    pass
+
+
+class State:
+    pass
+
+
+class NOMINAL(State):
+    pass
+
+
+class DEGRADED(State):
+    pass
+
+
+class CRITICAL(State):
+    pass
+
+
+class PowerHealth:
+    def __init__(
+        self,
+        logger: Logger,
+        config: Config,
+        power_monitor: PowerMonitorProto,
+    ) -> None:
+        self.logger: Logger = logger
+        self.config: Config = config
+        self._sensor: PowerMonitorProto = power_monitor
+
+    def get(self) -> NOMINAL | DEGRADED | CRITICAL:
+        """
+        Get the power health
+        """
+        errors: List[str] = []
+        self.logger.debug("Power monitor: ", sensor=self._sensor)
+
+        if isinstance(self._sensor, PowerMonitorProto):
```

Comment:
No longer need the isinstance check since we know `self._sensor` is of type `PowerMonitorProto`